visearch-android/src/main/java/com/visenze/visearch/android/network/SearchService.java

```diff
@@ -0,0 +1,186 @@
+package com.visenze.visearch.android.network;
+
+import com.visenze.visearch.android.ResultList;
+import com.visenze.visearch.android.UploadSearchParams;
+import com.visenze.visearch.android.ViSearch;
+import com.visenze.visearch.android.ViSearchException;
+import com.visenze.visearch.android.data.ResponseData;
+import com.visenze.visearch.android.util.AuthGenerator;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import okhttp3.MediaType;
+import okhttp3.MultipartBody;
+import okhttp3.RequestBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class SearchService {
+
+    public final static String ID_SEARCH = "search";
+    public final static String RECOMMENDATION = "recommendation";
+    public final static String COLOR_SEARCH = "colorsearch";
+    public final static String UPLOAD_SEARCH = "uploadsearch";
+    public final static String DISCOVER_SEARCH = "discoversearch";
+
+    private String appKey;
+    private String secretKey;
+    private String userAgent;
+
+
+    private APIService apiService;
+    public SearchService(String endPoint, String appKey, String secretKey, String userAgent) {
```

Comment:
missing authentication by only appKey (which is the access key). There are 2 ways to authenticate ViSearch:
- by key pair (access + secret)
- by only access key (provided that the access key is not the admin key)

Reply:
just saw the code below, can you create another constructor without the secret key and then call this one:

`this(endPoint, appKey, null, userAgent)`

Reply:
added already.